elastic/beats
# Invoke m.Run on TestMain and fix test flakiness

## Proposed commit message

```
invoke m.Run on TestMain and reduce test flakiness

Some TestMain were not running any tests as they did not invoke m.Run. This change fixes that.

Also other tests were presenting some flakyness, therefore this change also attempts to fix the flakiness by setting really unique IDs for filestream tests and increasing a timeout for TestTranslateGUIDWithLDAP.
```


The following 4 packages were not running any test as the `TestMain` was not invoking `m.Run()` to actually run the test. The packages were:
 - `filebeat/autodiscover/builder/hints`
 - `filebeat/processor/add_kubernetes_metadata`
 - `metricbeat/autodiscover/appender/kubernetes/token`
 - `metricbeat/autodiscover/builder/hints`


## Checklist




## Disruptive User Impact

 - N/A

## How to test this PR locally

Run the test for the affected packages, the tests should run.

```
go test -v ./filebeat/autodiscover/builder/hints
go test -v ./filebeat/processor/add_kubernetes_metadata
go test -v ./metricbeat/autodiscover/appender/kubernetes/token
go test -v ./metricbeat/autodiscover/builder/hints
```

Before the fix:
```
❯ go test -v .
ok  	github.com/elastic/beats/v7/filebeat/autodiscover/builder/hints	0.530s
```

After the fix
```
❯ go test -v .
=== RUN   TestUnpackCopiesDefault
--- PASS: TestUnpackCopiesDefault (0.00s)
=== RUN   TestGenerateHints
--- PASS: TestGenerateHints (0.01s)
=== RUN   TestGenerateHintsWithPaths
--- PASS: TestGenerateHintsWithPaths (0.00s)
PASS
ok  	github.com/elastic/beats/v7/filebeat/autodiscover/builder/hints	0.172s
```

## Related issues


- N/A
